### PR TITLE
migration/add-table-entity-pane-to-react

### PR DESCRIPTION
**Pull request by sunilyadav840**
Created AddTableEntityPanel/index.tsx (React component for AddTableEntityPanel panel). 
This PR includes both tableEntityPanel and CassenraRowPanel.
And cleanup knockout AddTableEntityPane.

https://user-images.githubusercontent.com/79906609/115373084-267b5180-a1e9-11eb-8c7f-a74f494fd397.mp4

### CODE REVIEW

File: src/Common/TableEntity/index.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import {
```
Comment (southpolesteve) [inline src/Common/TableEntity/index.tsx:1]: Change name from index.tsx

File: src/Common/TableEntity/index.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        <Dropdown
+          label={entityTypeLabel && entityTypeLabel}
+          selectedKey={selectedKey}
+          onChange={onEntityTypeChange}
+          options={options}
+          disabled={isPropertyTypeDisable}
+          id="entityTypeId"
+          styles={dropdownStyles}
+        />
+        {renderEntityValue()}
```
Comment (southpolesteve) [inline src/Common/TableEntity/index.tsx:150]: <EntityValue/ >

File: src/Common/TableEntity/index.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        className="addEntityTextField"
+        id="entityValueId"
+        autoFocus
+        type={getEntityValue()}
+        placeholder={entityValuePlaceholder}
+        value={typeof entityValue === "string" && entityValue}
+        onChange={onEntityValueChange}
+      />
+    );
+  };
```
Comment (southpolesteve) [inline src/Common/TableEntity/index.tsx:126]: Make this a component: `<EntityValue/>`

File: src/Explorer/Panes/Tables/AddTableEntityPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    } else {
+      const entityItems = tableEntityListViewModel.items();
+      const entityTypes = Utilities.getDataTypesFromEntities(headers, entityItems);
+      const defaultEntities: EntityRowType[] = getDefaultEntities(headers, entityTypes);
+      setEntities(defaultEntities);
+    }
+  }, []);
+
+  /* Add new entity attribute */
+  const submit = (event: React.FormEvent<HTMLInputElement>): void => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/Tables/AddTableEntityPanel.tsx:108]: Make an async function

File: src/Explorer/Panes/Tables/AddTableEntityPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    if (!isValidEntities(entities)) {
+      return undefined;
+    }
+    event.preventDefault();
+
+    const entity: Entities.ITableEntity = entityFromAttributes(entities);
+    explorer.tableDataClient
+      .createDocument(queryTablesTab.collection, entity)
+      .then((newEntity: Entities.ITableEntity) => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/Tables/AddTableEntityPanel.tsx:117]: Switch to async/await

File: src/Explorer/Panes/Tables/AddTableEntityPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+              </Stack>
+            )}
+          </div>
+          {renderPanelFooter()}
+        </div>
+      </form>
+    );
+  };
+
+  const renderPanelFooter = (): JSX.Element => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/Tables/AddTableEntityPanel.tsx:269]: `<PanelFooter/>`

File: src/Explorer/Panes/Tables/AddTableEntityPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+              );
+            })}
+            {userContext.apiType !== "Cassandra" && (
+              <Stack horizontal onClick={addNewEntity} className="addButtonEntiy">
+                <Image {...imageProps} src={AddPropertyIcon} alt="Add Entity" />
+                <Text className="addNewParamStyle">{getAddButtonLabel(userContext.apiType)}</Text>
+              </Stack>
+            )}
+          </div>
+          {renderPanelFooter()}
```
Comment (southpolesteve) [inline src/Explorer/Panes/Tables/AddTableEntityPanel.tsx:263]: `<PanelFooter/>`

**Review by southpolesteve — CHANGES_REQUESTED** [review]

File: src/Common/TableEntity/index.tsx (hunk 1 above)
Comment (sunilyadav840) [reply src/Common/TableEntity/index.tsx]: Updated

File: src/Common/TableEntity/index.tsx (hunk 2 above)
Comment (sunilyadav840) [reply src/Common/TableEntity/index.tsx]: Updated

File: src/Common/TableEntity/index.tsx (hunk 3 above)
Comment (sunilyadav840) [reply src/Common/TableEntity/index.tsx]: Updated

File: src/Explorer/Panes/Tables/AddTableEntityPanel.tsx (hunk 7 above)
Comment (sunilyadav840) [reply src/Explorer/Panes/Tables/AddTableEntityPanel.tsx]: Updated

File: src/Explorer/Panes/Tables/AddTableEntityPanel.tsx (hunk 5 above)
Comment (sunilyadav840) [reply src/Explorer/Panes/Tables/AddTableEntityPanel.tsx]: Updated code

**Review by southpolesteve — APPROVED** [review]

**Pull request merged**